alleycats-tests/shared/src/test/scala/alleycats/tests/ReferentialEqSuite.scala

```diff
@@ -3,42 +3,9 @@ package alleycats.tests
 import alleycats.ReferentialEq
 import cats.kernel.Eq
 import cats.kernel.laws.discipline._
-import cats.kernel.laws.EqLaws
-import org.scalacheck.Arbitrary
-import org.scalacheck.Gen
-import org.scalacheck.Prop.forAll
-import org.typelevel.discipline.Laws
 
 class ReferentialEqSuite extends AlleycatsSuite {
-
-  class ReferentialEqTests[A](eq: Eq[A]) extends Laws {
```

Comment:
Wait a minute: where did this go? Referential eq doesn't pass all the laws of `Eq`.

Reply:
See https://github.com/typelevel/cats/pull/3925#discussion_r669111963

Reply:
@armanbilge I may have messed something up but it seem to me that all laws have passed now. I see them all being green on my branch.

Reply:
@johnynek can probably explain it better than me, but IIUC:

Essentially, the anti-symmetry law states that `a === b <=> f(a) === f(b)` where `f` is a pure function. The definition of a pure function is that it is referentially transparent i.e. `f(a) === f(a)`. But, because we are implementing referential equality this may not hold. Example:
```scala
def prepend1(list: List[Int]) = 1 :: list

prepend1(Nil) == prepend1(Nil) // true
prepend1(Nil) eq prepend1(Nil) // false
```

Thus, we drop this law.

I'm not sure how/why the test suite is passing, but I suspect something is not quite right.